static/js/snap-metrics.js

```diff
@@ -0,0 +1,285 @@
+/* global d3, bb, moment */
+
+(function (window, d3, bb, moment) {
+  /**
+   * Debounce
+   * @param {Function} func Function to run.
+   * @param {Number} wait Time to wait between tries.
+   * @param {Boolean} immediate Immediately call func.
+   */
+  function debounce(func, wait, immediate) {
```

Comment:
This seems like something that could be useful in other places as well.
So the question is how to best share it across different modules...

Probably not worth to block this PR on this decision, but something to think of sooner than later.

Reply:
Yeah I think a lib of some kind will definitely be required moving forwards.